### PR TITLE
[core] Set 1 decimal precision for high distances lower than 10.0

### PR DESCRIPTION
**Pull request by gpesquero**
After the implementation of Distance class, the GetDistanceString() does not always return strings of high distances (km or miles) with 1 decimal precision.

For example, a distance of 5 km is returned as "5" where it should be "5.0" for a similar behavior with values such as "5.1" or "4.9".

This PR proposes an implementation to force 1 decimal precision for all high distance values below 10.0.

### CODE REVIEW

**Review by biodranik — COMMENTED** [review]
What is the behavior in other nav apps? Showing zero is not helpful, it could be an improvement over previous behavior.

**Comment by gpesquero** [comment]
Here's the behavior that I've noticed in other nav apps that I use:

**OsmAnd**
 - For distances >= 100 Km: No decimals (100 Km - 101 Km - ...)
 - For distances >= 10 Km:  Uses always 1 decimal precision for any value (90.1 Km - 90.0 Km - 89.9 Km)
 - For distances < 10 Km:   Uses always 2 decimal precision for any value (5.01 Km - 5.00 Km - 4.99 Km)

**Waze**
 - For distances >= 10 Km: No decimals (10 Km - 100 Km - ...)
 - For distances < 10 Km:  Uses 1 decimal precision, but not for all values (9.1 Km - 9 Km - 8.9 Km) 

**Google Maps**
 - For distances >= 10 Km: No decimals (10 Km - 100 Km - ...)
 - For distances < 10 Km:  Uses always 1 decimal precision (9.1 Km - 9.0 Km - 8.9 Km) 

For all the different nav apps, the stated behavior is the same for the display of the "next turn" distance as well as for the display of the "total remaining" distance.

**Comment by gpesquero** [comment]
By using a fixed precision setting for all high distances below 10.0 would also avoid the display of search results with different format, as it happens now:
 
imagen

File: platform/distance.cpp [hunk 1]
```diff
@@ -131,8 +131,16 @@ std::string Distance::GetDistanceString() const
   if (!IsValid())
     return "";
 
+  // Default precision is 0 (no decimals).
+  int precision = 0;
+
+  // Set 1 decimal precision for high distances (km, miles) lower than 10.0 (9.5, 7.0,...).
+  if ((m_distance < 10.0) && ((m_units == Units::Kilometers) || (m_units == Units::Miles)))
```
Comment (AndrewShkrob) [inline platform/distance.cpp:138]:
```suggestion
  if (m_distance < 10.0 && IsHighUnits())
```

File: platform/distance.cpp [hunk 2]
```diff
@@ -131,8 +131,16 @@ std::string Distance::GetDistanceString() const
...
+      precision = 1;
+
   std::ostringstream os;
-  os << std::defaultfloat << m_distance;
+  os.precision(precision);
+  os << std::fixed << m_distance;
```
Comment (AndrewShkrob) [inline platform/distance.cpp:143]:
```suggestion
  os << std::fixed << std::setprecision(precision) << m_distance;
```

Comment (gpesquero) [reply platform/distance.cpp]: Thanks @AndrewShkrob for your review. Commit updated accordingly...

**Review by biodranik — APPROVED** [review]
Ok, let's go for consistency :) Thanks!

**Pull request merged**